inst/plumber/v1/endpoints.R

```diff
@@ -406,6 +406,24 @@ function(req) {
   lkups$versions_paths[[req$argsQuery$version]]$cache_data_id
 }
 
+#* Retrieve grouped data stats
+#* @get /api/v1/grouped-stats
+#* @param welfare:[dbl] numeric vector for welfare
+#* @param population:[dbl] numeric vector for population
+#* @param requested_mean:[dbl] mean value
+#* @param povline:[dbl] poverty line value
+#* @param default_ppp:[dbl] default value for ppp
+#* @serializer unboxedJSON
+function(req) {
+  params <- req$args
+  # Split comma separated string as vector
+  params$welfare <- strsplit(params$welfare, ",")[[1]]
+  params$population <- strsplit(params$population, ",")[[1]]
+  params <- lapply(params, as.numeric)

```

Comment:
They are potentially many things that can go wrong here. Passing too many values for welfare or population, passing different number of value for welfare and population, etc.

It would be better to have specific function to parse these parameters, so we can unit test them for edge cases and ensure they work as expected.

Reply:
I Think that passing too many values is not a problem. Yet, I think a limit (say 100obs) suffices for the sake of the completeness of the functions. However, the check that both, welfare and population vectors should be the same length is necessary (though I think that it is implemented already in `wbpip`).

Reply:
Thanks @randrescastaneda We will still need some higher level validation so the API can return an informative to the end-users. Otherwise, they will just get uninformative 500 errors.

Reply:
I agree @tonyfujs , I have added a function to check inputs and return an informative error. Please have another look.